src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MetadataHelper.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+namespace System.Reflection.Emit
+{
+    // This static helper class adds common entities to a Metadata Builder.
+    internal static class MetadataHelper
+    {
+        internal static AssemblyReferenceHandle AddAssemblyReference(Assembly assembly, MetadataBuilder metadata)
+        {
+            AssemblyName assemblyName = assembly.GetName();
+
+            if (assemblyName == null || assemblyName.Name == null)
+            {
+                throw new ArgumentNullException(nameof(assemblyName));
+            }
+
+            return AddAssemblyReference(metadata, assemblyName.Name, assemblyName.Version, assemblyName.CultureName, assemblyName.GetPublicKey(), (AssemblyFlags)assemblyName.Flags);
+        }
+
+        internal static AssemblyReferenceHandle AddAssemblyReference(MetadataBuilder metadata, string name, Version? version, string? culture, byte[]? publicKey, AssemblyFlags flags)
+        {
+            return metadata.AddAssemblyReference(
+                name: metadata.GetOrAddString(name),
+                version: version ?? new Version(0, 0, 0, 0),
+                culture: (culture == null) ? default : metadata.GetOrAddString(value: culture),
+                publicKeyOrToken: (publicKey == null) ? default : metadata.GetOrAddBlob(publicKey),
+                flags: flags,
+                hashValue: default); // not sure where to find hashValue.
+        }
+
+        internal static TypeDefinitionHandle AddTypeDef(MetadataBuilder metadata, TypeBuilderImpl typeBuilder, EntityHandle baseType, int methodToken, int fieldToken)
```

Comment:
```suggestion
        internal static TypeDefinitionHandle AddTypeDefinition(MetadataBuilder metadata, TypeBuilderImpl typeBuilder, EntityHandle baseType, int methodToken, int fieldToken)
```
Consistent naming style withing the type